curator-x-discovery/src/main/java/org/apache/curator/x/discovery/details/ServiceCacheImpl.java

```diff
@@ -66,6 +68,30 @@
         cache.getListenable().addListener(this);
     }
 
+    ServiceCacheImpl(ServiceDiscoveryImpl<T> discovery, String name, ExecutorService executorService)
```

Comment:
I'd prefer to see 1 ctor. It can take a CloseableExecutorService. The builder can wrap the executorService. I think there are other examples of this in the Curator code.